### ISSUE TITLE
Ensure object returned from GitVersion is consistent

### ISSUE THREAD

**Issue by Jericho** [cake-build/cake#1128]
I have observed that the object returned by the GitVersion alias is different depending on the OutputType settings property. If set the OutputType to 'GitVersionOutput.BuildServer', GitVersion returns an object where the properties contain NULL but if I omit the OutputType setting, the properties on the returned object are properly initialized.

It was suggested to me to invoke the GitVersion alias twice to work around this issue but I still think the value returned by this alias should be consistent. At the very least, the documentation should be updated to warn people about this behavior.

For more details including sample code to reproduce, see this [StackOverflow question](http://stackoverflow.com/questions/38603170/object-returned-from-gitversion-not-consistent/38605556#38605556)

**Comment by gep13** [comment]
@Jericho I agree that we need to do something, however, we have to remember that Cake is simply a wrapper around the underlying tool.  If you run `gitversion` directly using the buildserver option, you won't get any JSON output.

**Comment by gep13** [comment]
@Jericho in all likelihood, we won't be able to do anything with this until we implement a new feature in GitVersion itself: https://github.com/GitTools/GitVersion/issues/983

Once that is done, we will modify the GitVersion alias in Cake to take advantage of the new file that is generated.

**Comment by asbjornu** [comment]
Once implemented, #1030 will make sure that the output from GitVersion is consistent and don't regress.
